### ISSUE TITLE
SIGSEGV when switching on I2PControlService

### ISSUE THREAD

**Issue by anonimal** [monero-project/kovri#63]
```
-- Performing Test CXX11_SUPPORTED
-- Performing Test CXX11_SUPPORTED - Success
-- Performing Test CXX0X_SUPPORTED
-- Performing Test CXX0X_SUPPORTED - Success
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE  
-- Boost version: 1.60.0
-- Found the following Boost libraries:
--   chrono
--   log
--   program_options
--   date_time
--   thread
--   system
--   filesystem
--   regex
-- Found Crypto++: /usr/include, /usr/lib64/libcryptopp.so
-- Found OpenSSL: /usr/lib64/libssl.so;/usr/lib64/libcrypto.so (found version "1.0.2e") 
-- Found MiniUPnP headers: /usr/include/miniupnpc
-- ---------------------------------------
-- Build type         : Debug
-- Compiler vendor    : GNU
-- Compiler version   : 5.3.0
-- Compiler path      : /usr/bin/c++
-- Install prefix:    : /usr/local
-- Kovri data directory: /kovri/.kovri
-- Options:
--   AESNI            : OFF
--   BENCHMARKS       : OFF
--   BINARY           : ON
--   BOOST LOG        : ON
--   DOXYGEN          : OFF
--   HARDENING        : OFF
--   LIBRARY          : ON
--   OPTIMIZATION     : OFF
--   STATIC BUILD     : OFF
--   TESTS            : OFF
--   UPnP             : OFF
-- ---------------------------------------
```
```
(gdb) run --i2pcontrolport 7650
...
[Thread 0x7fffdffff700 (LWP 25414) exited]
2016:01:10|12:40:56.993849 default:default              NFO             Starting I2PControlService ...

Program received signal SIGSEGV, Segmentation fault.
0x00007ffff536aac4 in pthread_mutex_lock () from /usr/lib/libpthread.so.0
(gdb) bt
#0  0x00007ffff536aac4 in pthread_mutex_lock () from /usr/lib/libpthread.so.0
#1  0x000000000096d408 in boost::asio::detail::posix_mutex::lock (this=0x2d2d2d2d2d2d2d35) at /usr/include/boost/asio/detail/posix_mutex.hpp:52
#2  0x000000000097032e in boost::asio::detail::scoped_lock<boost::asio::detail::posix_mutex>::scoped_lock (this=0x7fffffffd600, m=...) at /usr/include/boost/asio/detail/scoped_lock.hpp:46
#3  0x000000000096d6a0 in boost::asio::detail::service_registry::do_use_service (this=0x2d2d2d2d2d2d2d2d, key=..., 
    factory=0x97c294 <boost::asio::detail::service_registry::create<boost::asio::deadline_timer_service<boost::posix_time::ptime, boost::asio::time_traits<boost::posix_time::ptime> > >(boost::asio::io_service&)>)
    at /usr/include/boost/asio/detail/impl/service_registry.ipp:108
#4  0x000000000097af96 in boost::asio::detail::service_registry::use_service<boost::asio::deadline_timer_service<boost::posix_time::ptime, boost::asio::time_traits<boost::posix_time::ptime> > > (this=0x2d2d2d2d2d2d2d2d)
    at /usr/include/boost/asio/detail/impl/service_registry.hpp:48
#5  0x0000000000978ce1 in boost::asio::use_service<boost::asio::deadline_timer_service<boost::posix_time::ptime, boost::asio::time_traits<boost::posix_time::ptime> > > (ios=...) at /usr/include/boost/asio/impl/io_service.hpp:33
#6  0x0000000000975f6f in boost::asio::basic_io_object<boost::asio::deadline_timer_service<boost::posix_time::ptime, boost::asio::time_traits<boost::posix_time::ptime> >, false>::basic_io_object (this=0xecc700, io_service=...)
    at /usr/include/boost/asio/basic_io_object.hpp:90
#7  0x0000000000971a5f in boost::asio::basic_deadline_timer<boost::posix_time::ptime, boost::asio::time_traits<boost::posix_time::ptime>, boost::asio::deadline_timer_service<boost::posix_time::ptime, boost::asio::time_traits<boost::posix_time::ptime> > >::basic_deadline_timer (this=0xecc700, io_service=...) at /usr/include/boost/asio/basic_deadline_timer.hpp:149
#8  0x00000000009f2392 in i2p::client::i2pcontrol::I2PControlSession::I2PControlSession (this=0xecc5b0, ios=..., pass="itoopie") at /kovri/src/api/I2PControl/I2PControl.cpp:203
#9  0x00000000009b3bea in __gnu_cxx::new_allocator<i2p::client::i2pcontrol::I2PControlSession>::construct<i2p::client::i2pcontrol::I2PControlSession, boost::asio::io_service&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&> (this=0x7fffffffde97, __p=0xecc5b0) at /usr/include/c++/5.3.0/ext/new_allocator.h:120
#10 0x00000000009b2d20 in std::allocator_traits<std::allocator<i2p::client::i2pcontrol::I2PControlSession> >::_S_construct<i2p::client::i2pcontrol::I2PControlSession, boost::asio::io_service&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&> (__a=..., __p=0xecc5b0) at /usr/include/c++/5.3.0/bits/alloc_traits.h:256
#11 0x00000000009b184a in std::allocator_traits<std::allocator<i2p::client::i2pcontrol::I2PControlSession> >::construct<i2p::client::i2pcontrol::I2PControlSession, boost::asio::io_service&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&> (__a=..., __p=0xecc5b0) at /usr/include/c++/5.3.0/bits/alloc_traits.h:402
#12 0x00000000009afb34 in std::_Sp_counted_ptr_inplace<i2p::client::i2pcontrol::I2PControlSession, std::allocator<i2p::client::i2pcontrol::I2PControlSession>, (__gnu_cxx::_Lock_policy)2>::_Sp_counted_ptr_inplace<boost::asio::io_service&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&> (this=0xecc5a0, __a=...) at /usr/include/c++/5.3.0/bits/shared_ptr_base.h:522
#13 0x00000000009adf22 in std::__shared_count<(__gnu_cxx::_Lock_policy)2>::__shared_count<i2p::client::i2pcontrol::I2PControlSession, std::allocator<i2p::client::i2pcontrol::I2PControlSession>, boost::asio::io_service&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&> (this=0xec8da8, __a=...) at /usr/include/c++/5.3.0/bits/shared_ptr_base.h:617
#14 0x00000000009ac6e7 in std::__shared_ptr<i2p::client::i2pcontrol::I2PControlSession, (__gnu_cxx::_Lock_policy)2>::__shared_ptr<std::allocator<i2p::client::i2pcontrol::I2PControlSession>, boost::asio::io_service&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&> (this=0xec8da0, __tag=..., __a=...) at /usr/include/c++/5.3.0/bits/shared_ptr_base.h:1097
#15 0x00000000009aaeb6 in std::shared_ptr<i2p::client::i2pcontrol::I2PControlSession>::shared_ptr<std::allocator<i2p::client::i2pcontrol::I2PControlSession>, boost::asio::io_service&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&> (this=0xec8da0, __tag=..., __a=...) at /usr/include/c++/5.3.0/bits/shared_ptr.h:319
#16 0x00000000009a7769 in std::allocate_shared<i2p::client::i2pcontrol::I2PControlSession, std::allocator<i2p::client::i2pcontrol::I2PControlSession>, boost::asio::io_service&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&> (__a=...) at /usr/include/c++/5.3.0/bits/shared_ptr.h:614
#17 0x00000000009a4a67 in std::make_shared<i2p::client::i2pcontrol::I2PControlSession, boost::asio::io_service&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&> ()
    at /usr/include/c++/5.3.0/bits/shared_ptr.h:630
#18 0x000000000099ecad in i2p::client::i2pcontrol::I2PControlService::I2PControlService (this=0xec8da0, address="127.0.0.1", port=7650, password="itoopie") at /kovri/src/api/I2PControl/I2PControlServer.cpp:63
#19 0x000000000094f844 in i2p::client::ClientContext::Start (this=0xe78b80 <i2p::client::context>) at /kovri/src/client/ClientContext.cpp:140
#20 0x000000000093987a in i2p::util::Daemon_Singleton::Start (this=0xe78820 <i2p::util::DaemonLinux::Instance()::instance>) at /kovri/src/client/Daemon.cpp:121
#21 0x000000000094e025 in i2p::util::DaemonLinux::Start (this=0xe78820 <i2p::util::DaemonLinux::Instance()::instance>) at /kovri/src/client/DaemonLinux.cpp:127
#22 0x0000000000938597 in main (argc=3, argv=0x7fffffffe8c8) at /kovri/src/Kovri.cpp:46
(gdb)
```

**Comment by anonimal** [comment]
**Edit**: added system/build information.

**Comment by majestrate** [comment]
fixed in https://github.com/majestrate/kovri/commit/39b513a58b2f85b63eb38e5f1e9a5f78c3a53157

**Comment by majestrate** [comment]
that commit however also has changes related to #61 but not completed (yet)

**Comment by anonimal** [comment]
Fixed in 39b513a58b2f85b63eb38e5f1e9a5f78c3a53157. Merged in 6d8d82e43f71440e01bd41efbe31aa2be3bb9137
